OpenRA.Game/FileFormats/Png.cs

```diff
@@ -136,14 +137,34 @@ public Png(Stream s)
 								{
 									var pxStride = PixelStride;
 									var rowStride = Width * pxStride;
+									var pixelsPerByte = 8 / bitDepth;
+									var sourceRowStride = Exts.IntegerDivisionRoundingAwayFromZero(rowStride, pixelsPerByte);
 
 									Span<byte> prevLine = new byte[rowStride];
 									for (var y = 0; y < Height; y++)
 									{
 										var filter = (PngFilter)ds.ReadUInt8();
-										ds.ReadBytes(Data, y * rowStride, rowStride);
+										ds.ReadBytes(Data, y * rowStride, sourceRowStride);
 										var line = Data.AsSpan(y * rowStride, rowStride);
 
+										// If the source has a bit depth of 1, 2 or 4 it packs multiple pixels per byte.
```

Comment:
Suggestion to wrap this whole method in a top level try catch that will clarify which png image was read during exception.

i.e. in cases where the png data is corrupt.

Reply:
The input is a stream, so there's no value the PNG class can add - it has no access to filenames for example. The callers are the ones that will need to wrap the exceptions with extra context, if they want to provide any.